app/controllers/support/schools/devices/order_status_controller.rb

```diff
@@ -1,88 +1,120 @@
-          service.update!
-        end
-        flash[:success] = t(:success, scope: %i[support order_status update])
-        redirect_to support_school_path(urn: @school.urn)
-      else
-        redirect_to support_school_confirm_enable_orders_path(urn: @school.urn,
-                                                              order_state: @form.order_state,
-                                                              laptop_cap: @form.laptop_cap,
-                                                              router_cap: @form.router_cap)
-      end
-    else
-      render :edit, status: :unprocessable_entity
-    end
+    form.save(validate: false) && updated!
   rescue Computacenter::OutgoingAPI::Error => e
-    flash[:warning] = t(:cap_update_request_error, scope: %i[support order_status update], payload_id: e.cap_update_request&.payload_id)
-    render :edit, status: :unprocessable_entity
+    update_error!(e.cap_update_request&.payload_id)
   end
 
   # GET /support/devices/schools/:urn/enable-orders/confirm
   def confirm
-    @form = Support::EnableOrdersForm.new(order_state: params[:order_state],
-                                          laptop_cap: params[:laptop_cap],
-                                          router_cap: params[:router_cap])
-    @laptop_allocation = @school.laptop_allocation
-    @router_allocation = @school.router_allocation
+    set_enable_orders_form(confirm_params)
+    @laptop_allocation = school.laptop_allocation
+    @router_allocation = school.router_allocation
   end
 
   def collect_urns_to_allow_many_schools_to_order
-    authorize School, :edit?
-    @form = Support::BulkAllocationForm.new
+    set_bulk_allocation_form
   end
 
   def allow_ordering_for_many_schools
+    set_bulk_allocation_form(**restriction_params)
+    form.save ? ordering_allowed_for_many_schools! : error_allowing_ordering_for_many_schools!
+  end
+
+private
+
+  def set_bulk_allocation_form(**params)
+    @form = Support::BulkAllocationForm.new(**params)
+  end
+
+  # Filters
+  def authorize_school_edition
     authorize School, :edit?
-    @form = Support::BulkAllocationForm.new(restriction_params)
+  end
 
-    if @form.valid?
-      importer = Importers::AllocationUploadCsv.new(
-        path_to_csv: @form.upload.path,
-        send_notification: @form.send_notification,
-      )
-      importer.call
-      redirect_to support_allocation_batch_job_path(importer.batch_id)
-    else
-      render :collect_urns_to_allow_many_schools_to_order, status: :unprocessable_entity
-    end
+  def check_confirmation
+    unconfirmed! if params[:confirm].blank?
   end
 
-private
+  def set_enable_orders_form(updates = enable_orders_form_params)
+    @form = Support::EnableOrdersForm.new(school: school, **updates)
+  end
 
   def set_school
     @school = School.where_urn_or_ukprn_or_provision_urn(params[:school_urn]).first!
-    authorize @school, :show?
+    authorize school, :show?
```

Comment:
Is this definitely right to use `school` instead of `@school`?

Reply:
Yes, because there is `attr_reader :school` at the top of the file that basically defines
```
def school
   @school 
end
```

Reply:
Sorry didn't see that